Webpack plugin errors do not crash the build

As first reported with in #1510, here is an example I encountered with the Webpack Plugin pre #1525
```
[@snowpack/plugin-webpack] optimize() starting…

./index.js
Module build failed (from ../node_modules/babel-loader/lib/index.js):
Error: Cannot find module '@babel/preset-env' from '/Users/melissamcewen/Sites/my-first-snowpack/build'
    at Function.resolveSync [as sync] (/Users/melissamcewen/Sites/my-first-snowpack/node_modules/resolve/lib/sync.js:89:15)
 3 assets
[@snowpack/plugin-webpack] ✔ optimize() success
```

The build failed but it reported success. 

CC @denis-sokolov

--- a/plugins/plugin-webpack/README.md
+++ b/plugins/plugin-webpack/README.md
@@ -57,6 +57,7 @@ The options object is optional.
 - `extendConfig: (config: WebpackConfig) => WebpackConfig` - extend your webpack config, see below.
 - `manifest: boolean | string` - Enable generating a manifest file. The default value is `false`, the default file name is `./asset-manifest.json` if setting manifest to `true`. The relative path is resolved from the output directory.
 - `htmlMinifierOptions: boolean | object` - [See below](#minify-html).
+- `ignoreWarnings: boolean` - Does not fail the build when Webpack emits warnings. The default value is `true`.
 
 #### Extending The Default Webpack Config
 

--- a/plugins/plugin-webpack/README.md
+++ b/plugins/plugin-webpack/README.md
@@ -57,7 +57,7 @@ The options object is optional.
 - `extendConfig: (config: WebpackConfig) => WebpackConfig` - extend your webpack config, see below.
 - `manifest: boolean | string` - Enable generating a manifest file. The default value is `false`, the default file name is `./asset-manifest.json` if setting manifest to `true`. The relative path is resolved from the output directory.
 - `htmlMinifierOptions: boolean | object` - [See below](#minify-html).
-- `ignoreWarnings: boolean` - Does not fail the build when Webpack emits warnings. The default value is `true`.
+- `failOnWarnings: boolean` - Does fail the build when Webpack emits warnings. The default value is `false`.
 
 #### Extending The Default Webpack Config
 

--- a/plugins/plugin-webpack/plugin.js
+++ b/plugins/plugin-webpack/plugin.js
@@ -199,8 +199,6 @@ module.exports = function plugin(config, args = {}) {
       ? './asset-manifest.json'
       : undefined;
 
-  const ignoreWarnings = args.ignoreWarnings === undefined ? true : args.ignoreWarnings
-
   // Webpack handles minification for us, so its safe to always
   // disable Snowpack's default minifier.
   config.buildOptions.minify = false;
@@ -363,11 +361,16 @@ module.exports = function plugin(config, args = {}) {
             reject(err);
             return;
           }
-          if (stats.hasErrors() || (stats.hasWarnings() && !ignoreWarnings)) {
-            const info = stats.toJson(extendedConfig.stats);
-            console.error(info.warnings.join('\n-----\n'));
-            console.error(info.errors.join('\n-----\n'));
-            reject(Error(`Webpack build failed with ${info.errors} error(s) and ${info.warnings} warning(s)`));
+          const info = stats.toJson(extendedConfig.stats);
+          if (stats.hasErrors()) {
+            console.error('Webpack errors:\n' + info.errors.join('\n-----\n'));
+            reject(Error(`Webpack failed with ${info.errors} error(s).`));
+          }
+          if (stats.hasWarnings()) {
+            console.error('Webpack warnings:\n' + info.warnings.join('\n-----\n'));
+            if (args.failOnWarnings) {
+              reject(Error(`Webpack failed with ${info.warnings} warnings(s).`));
+            }
           }
           resolve(stats);
         });

--- a/plugins/plugin-webpack/plugin.js
+++ b/plugins/plugin-webpack/plugin.js
@@ -365,11 +365,13 @@ module.exports = function plugin(config, args = {}) {
           if (stats.hasErrors()) {
             console.error('Webpack errors:\n' + info.errors.join('\n-----\n'));
             reject(Error(`Webpack failed with ${info.errors} error(s).`));
+            return;
           }
           if (stats.hasWarnings()) {
             console.error('Webpack warnings:\n' + info.warnings.join('\n-----\n'));
             if (args.failOnWarnings) {
               reject(Error(`Webpack failed with ${info.warnings} warnings(s).`));
+              return;
             }
           }
           resolve(stats);

--- a/plugins/plugin-webpack/plugin.js
+++ b/plugins/plugin-webpack/plugin.js
@@ -199,6 +199,8 @@ module.exports = function plugin(config, args = {}) {
       ? './asset-manifest.json'
       : undefined;
 
+  const ignoreWarnings = args.ignoreWarnings === undefined ? true : args.ignoreWarnings
+
   // Webpack handles minification for us, so its safe to always
   // disable Snowpack's default minifier.
   config.buildOptions.minify = false;
@@ -361,10 +363,11 @@ module.exports = function plugin(config, args = {}) {
             reject(err);
             return;
           }
-          if (stats.hasErrors()) {
+          if (stats.hasErrors() || (stats.hasWarnings() && !ignoreWarnings)) {
             const info = stats.toJson(extendedConfig.stats);
             console.error(info.warnings.join('\n-----\n'));
             console.error(info.errors.join('\n-----\n'));
+            reject(Error(`Webpack build failed with ${info.errors} error(s) and ${info.warnings} warning(s)`));
           }
           resolve(stats);
         });

